src/main/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipant.groovy

```diff
@@ -283,15 +287,17 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 
 		this.mavenSession = mavenSession
 
-		this.remoteRepositories = mavenSession.request.remoteRepositories
-		this.localRepository = mavenSession.request.localRepository
-
 		this.mavenVersionIsolate = discoverMavenVersion(mavenSession)
 
 		this.modelCache = new NotDefaultModelCache(mavenSession)
 
-		repositoryFactory = mavenSession.container.lookup(ArtifactRepositoryFactory)
-		repositoryLayouts = mavenSession.lookupMap(ArtifactRepositoryLayout.class.getName()) as Map<String, ArtifactRepositoryLayout>
+		// why are these looked up explicitly? seem to be injected just fine
```

Comment:
Probably looked up as we may have not known it could be injected, or we have issues with it being injected on different versions? Hard to recall now.

If they inject fine, we could probably kill the `if` check and legacy lookup..

Reply:
I tested with maven 3.6.0 and maven 3.3.9. I don't use eclipse with m2e so i don't know if this beaks that tool (I noticed that there are some comments about this combination)